src/bodse/timetable/download.py

```diff
@@ -0,0 +1,215 @@
+
+
+def download_gtfs_feed(directory):
+    """Function that will download the GTFS feed from BODS
+
+    Will download and unzip the lastest GTFS feed from BODS unless there is a feed less than a week old already in the specified repoistory.
+    Previous feeds will be moved to a "SS" folder.
+
+    Args:
+        directory (Path): filepath where GTFS feed is to be saved. Use full filepath rather then relative path.
+    """
+    current_datetime = datetime.datetime.now()
+    current_date_time = current_datetime.strftime("%Y%m%d_%H-%M-%S")
+
+    try:
+        LOG.info("Downloading started")
+
+        filename = Path(f"GTFS_{current_date_time}.zip")
+        filepath = directory / filename
+
+        download_to_file(filepath)
+
+        # Extract the downloaded zip file to a folder with the same name
+        extraction_folder = directory / filename.stem
+        extraction_folder.mkdir(exist_ok=True)
+
+        with zipfile.ZipFile(filepath, "r") as zip_ref:
+            zip_ref.extractall(extraction_folder)
+
+        LOG.info("Extracted contents to %s", extraction_folder)
+
+        # Create an "ss" folder and move previous files
+        ss_folder = directory / "ss"
+        ss_folder.mkdir(exist_ok=True)
+
+        for file in directory.glob("GTFS_*"):
+            if str(file.name) == str(filename):
+                continue
+            elif file != extraction_folder:
+                new_location = ss_folder / file.name
+                file.rename(new_location)
+                LOG.info("Moved %s to %s", file.name, new_location)
+
+    except requests.exceptions.RequestException:
+        LOG.error("Download failed:", exc_info=True)
+
+    except zipfile.BadZipFile:
+        LOG.error("Failed to unzip the downloaded file.", exc_info=True)
+
+    except Exception:
+        LOG.error("An error occurred:", exc_info=True)
+
+
+def gtfs_filename() -> str:
+    """Standard GTFS filename with todays date."""
+    return _FILENAME_FORMAT.format(date=datetime.date.today())
+
+
+def find_gtfs_files(folder: Path) -> dict[datetime.date, Path]:
+    """Find GTFS files with specific name format in `folder`."""
+    files: dict[datetime.date, Path] = {}
+
+    pattern = re.compile(_FILENAME_FORMAT_BASE.format(r"(\d{2})(\d{2})(\d{2})"), re.I)
+    for path in folder.glob(_FILENAME_FORMAT_BASE.format("*")):
+        match = pattern.match(path.name)
+        if match is None:
+            warnings.warn(f"found GTFS file with incorrect name format '{path.name}'")
+            continue
+
+        year, month, day = match.groups()
+        gtfs_date = datetime.date(int(year), int(month), int(day))
+        files[gtfs_date] = path
+
+    return files
+
+
+def get_feed_version(path: Path) -> Optional[datetime.datetime]:
+    """Attempt to get feed version from GTFS file, if available."""
+    if not path.is_file():
+        LOG.warning("cannot get feed version for non-existent file '%s'", path)
+        return None
+
+    if not zipfile.is_zipfile(path):
```

Comment:
The scheduler process which calls `get_feed_version` doesn't call `download_gtfs_feed` it calls `download_to_file` which just saves the GTFS zipfile to a folder and doesn't extract it. `download_to_file` isn't used as part of the scheduler process.

Reply:
Ahh cheers, that makes sense. 

Apologies, I was looking down at main()

I'll stop for the day to avoid any more confusion 😄, think I've read too many lines for one afternoon!